app/controllers/pensions/annualAllowance/PensionSchemeTaxReferenceController.scala

```diff
@@ -43,8 +43,23 @@ class PensionSchemeTaxReferenceController @Inject()(implicit val cc: MessagesCon
                                                     errorHandler: ErrorHandler,
                                                     clock: Clock) extends FrontendController(cc) with I18nSupport {
 
+  def formDetails(isAgent : Boolean) : (String, String) = {
+    if (isAgent) {
+      (
+        "pension.pensionSchemeTaxReference.error.noEntry.agent",
+        "pension.pensionSchemeTaxReference.error.incorrectFormat.agent"
+      )
+    } else {
+      (
+        "pension.pensionSchemeTaxReference.error.noEntry.individual",
+        "pension.pensionSchemeTaxReference.error.incorrectFormat.individual"
+      )
+    }
+  }
+
   private def prefillValue(pstrListOpt: Option[Seq[String]], pensionSchemeTaxReference: Option[Int], user: User): Form[String] = {
-    val emptyForm: Form[String] = PensionSchemeTaxReferenceForm.pensionSchemeTaxReferenceForm(user.isAgent)
+    val errorMsgDetails = formDetails(user.isAgent)
+    val emptyForm: Form[String] = PensionSchemeTaxReferenceForm.pensionSchemeTaxReferenceForm(errorMsgDetails._1, errorMsgDetails._2)
```

Comment:
Not needed to be so verbose, you can do something like this 
"(s"pension.pensionSchemeTaxReference.error.noEntry.${if (isAgent) "agent" else "individual"}" and then you could even do that inline and not have to write the function at all.

Reply:
It would be enough to have a val and interpolate that.